### PR TITLE
fix validation parameter and style

### LINKED ISSUE #22401: resource/aws_vpc_ipam_pool_cidr_allocation fix validation parameters for `netmask_length` and styling updates
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* aws_vpc_ipam_pool_cidr_allocation

A couple of misses were discovered in the big PR with various IPAM resources.
- `ValidateFunc` on netmask_length should match the API valid values, see [context](https://github.com/hashicorp/terraform-provider-aws/pull/22341/files#r775554658)
- CRUD function names were not updated to match the new service organizational structure. They are being unnecessarily exported: https://github.com/hashicorp/terraform-provider-aws/blob/f445f5e1980a8b4a6581b2964b8ba70b80330128/internal/service/ec2/vpc_ipam_pool_cidr_allocation.go#L20

### References

* https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/service-package-pullrequest-guide.md

### PR DESCRIPTION
**Pull request by rahul-amzn**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #22401

Output from acceptance testing:

```
$ make testacc TESTS=TestAccVPCIpamPoolAllocation PKG=ec2 ACCTEST_PARALLELISM=1     
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 1 -run='TestAccVPCIpamPoolAllocation' -timeout 180m
=== RUN   TestAccVPCIpamPoolAllocation_ipv4Basic
=== PAUSE TestAccVPCIpamPoolAllocation_ipv4Basic
=== RUN   TestAccVPCIpamPoolAllocation_ipv4BasicNetmask
=== PAUSE TestAccVPCIpamPoolAllocation_ipv4BasicNetmask
=== CONT  TestAccVPCIpamPoolAllocation_ipv4Basic
--- PASS: TestAccVPCIpamPoolAllocation_ipv4Basic (119.37s)
=== CONT  TestAccVPCIpamPoolAllocation_ipv4BasicNetmask
--- PASS: TestAccVPCIpamPoolAllocation_ipv4BasicNetmask (115.20s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/ec2        237.762s

...
```

### CODE REVIEW

**Comment by rahul-amzn** [comment]
@drewmullen @GlennChia can you please review? Thanks!

**Comment by rahul-amzn** [comment]
Proposing change to expand `cidr` and `netmask_length` attributes netmask to a minimum of 0 and maximum of 32.

File: .changelog/22418.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
```
Comment (drewmullen) [inline .changelog/22418.txt:1]: this can be `:bug` instead of `enhancement`

**Review by drewmullen — CHANGES_REQUESTED** [review]
2 changes proposed.

1. chagelog type
1. Can you add a note to the end the [attribute description](https://github.com/hashicorp/terraform-provider-aws/blob/c0284b85fbd2561a63b277e0f3351cee4e95ae24/website/docs/r/vpc_ipam_pool_cidr_allocation.html.markdown?plain=1#L53) detailing the valid values `0-32`? Example of how those looks [here](https://github.com/hashicorp/terraform-provider-aws/blob/c0284b85fbd2561a63b277e0f3351cee4e95ae24/website/docs/r/vpc_ipam_pool.html.markdown?plain=1#L80)

LGTM once those are done!

File: .changelog/22418.txt (hunk 1 above)
Comment (rahul-amzn) [reply .changelog/22418.txt]: Thanks @drewmullen, changes done.

**Review by drewmullen — APPROVED** [review]
LGTM!

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTS=TestAccVPCIpamPoolAllocation PKG=ec2 ACCTEST_PARALLELISM=1
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 1 -run='TestAccVPCIpamPoolAllocation' -timeout 180m
=== RUN   TestAccVPCIpamPoolAllocation_ipv4Basic
=== PAUSE TestAccVPCIpamPoolAllocation_ipv4Basic
=== RUN   TestAccVPCIpamPoolAllocation_ipv4BasicNetmask
=== PAUSE TestAccVPCIpamPoolAllocation_ipv4BasicNetmask
=== CONT  TestAccVPCIpamPoolAllocation_ipv4Basic
--- PASS: TestAccVPCIpamPoolAllocation_ipv4Basic (80.48s)
=== CONT  TestAccVPCIpamPoolAllocation_ipv4BasicNetmask
--- PASS: TestAccVPCIpamPoolAllocation_ipv4BasicNetmask (67.32s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	151.379s
```

**Pull request merged**